logstash-core/spec/logstash/config/mixin_spec.rb

```diff
@@ -137,6 +141,18 @@
       expect(subject.strings).to match_array(strings)
     end
 
+    context 'a URI list with entries containing whitespace-separated URIs' do
+      let(:config) { super().merge("uris" => uris.join(' ')) }
+      before(:each) do
+        # ensure that we actually instantiate it with our modified `config`,
+        # since `secure_params!` mutates the `original_params`.
+        expect(klass).to receive(:new).with(a_hash_including("uris" => uris.join(' '))).and_call_original
```

Comment:
I suggest placing this in a separate test if it's something we're not comfortable assuming that just works. otherwise we can flip this to an `allow`

Reply:
I've refactored the tests for list/URI, including coverage for the original case (array with many individual URI strings), for other cases that worked before (literal string with one URI), and for the new behaviour (literal string with many URIs, mixed arrays, etc.)